packages/medusa/src/services/product-variant.ts

```diff
@@ -255,9 +255,12 @@ class ProductVariantService extends BaseService {
       if (prices) {
         for (const price of prices) {
           if (price.region_id) {
+            const region = await this.regionService_.retrieve(price.region_id)
+
             await this.setRegionPrice(result.id, {
               amount: price.amount,
               region_id: price.region_id,
+              currency_code: region.currency_code,
```

Comment:
issue: this shouldn't be necessary as the currency is "inferred" by the region whenever the price is used

Reply:
It won't be able to create the MA record due to "currency_code not null constraint".
Please see [this](https://github.com/medusajs/medusa/pull/1706/files#diff-0eb545096b9448e2690437e89303259b7ed652c906ac5202edb4b6010a24aec4R1755) test case I added: without adding currency_code this way, the test will fail

Reply:
Alright, all good then - maybe we should revisit at a later point to figure out if it is better to allow prices without currency code where the currency is inferred by the region. But we can have that discussion some other time :)